### PR TITLE
306 relationship display fields (for all models)

### PR DESCRIPTION
**Pull request by lampwins**
### Fixes: #306 

This PR implements an opinionated method of exposing a standardized `display` field on all model serializers. It sources from the instance's `display` property and falls back to `__str__()`. This functionally negates the need for using `display_field` on the `DynamicModelChoiceField` within the core (as we stand today), but the functionality is left intact should the need ever arise. This also eases the burden on plugin and Job authors.

### CODE REVIEW

File: nautobot/core/api/__init__.py [hunk 1]
```diff
@@ -19,6 +20,7 @@
     "OrderedDefaultRouter",
     "SerializedPKRelatedField",
     "TimeZoneField",
+    "BaseModelSerializer",
```
Comment (glennmatthews) [inline nautobot/core/api/__init__.py:23]: alphabetical order please :-)

File: nautobot/core/api/serializers.py [hunk 2]
```diff
@@ -6,13 +6,46 @@
...
+    def get_field_names(self, declared_fields, info):
+        """
+        Override get_field_names() to append the `display` field so it is always included in the
+        serializer's `Meta.fields`.
+
+        DRF does not automatically add declared fields to `Meta.fields`, nor does it require that declared fields
+        on a super class be included in `Meta.fields` to allow for a subclass to include only a subset of declared
+        fields from the super. This means either we intercept and append the display field at this level, or
+        enforce by convention that all consumers of BaseModelSerializer include `display` in their `Meta.fields`
+        which would surely lead to errors of omition.
```
Comment (glennmatthews) [inline nautobot/core/api/serializers.py:40]:
```suggestion
        which would surely lead to errors of omission; therefore we have chosen the former approach.
```

File: nautobot/docs/additional-features/jobs.md [hunk 3]
```diff
@@ -168,16 +168,16 @@ Similar to `ChoiceVar`, but allows for the selection of multiple choices.
 A particular object within Nautobot. Each ObjectVar must specify a particular model, and allows the user to select one of the available instances. ObjectVar accepts several arguments, listed below.
 
 * `model` - The model class
-* `display_field` - The name of the REST API object field to display in the selection list (default: `'name'`)
+* `display_field` - The name of the REST API object field to display in the selection list (default: `'display'`)
 * `query_params` - A dictionary of query parameters to use when retrieving available options (optional)
 * `null_option` - A label representing a "null" or empty choice (optional)
 
-The `display_field` argument is useful when referencing a model which does not have a `name` field. For example, when displaying a list of device types, you would likely use the `model` field:
+The `display_field` argument is useful in cases where using the `display` API field is not desired for referencing the object. For example, when displaying a list of IP Addresses, you might want to use the `dns_name` field:
```
Comment (glennmatthews) [inline nautobot/docs/additional-features/jobs.md:175]: 👍  Thanks for revising the example!

File: nautobot/docs/installation/migrating-from-netbox.md [hunk 4]
```diff
@@ -71,6 +71,10 @@ The following backwards-incompatible changes have been made to the data model in
 
 A new [`Status`](https://nautobot.readthedocs.io/en/latest/models/extras/status/) model has been added to represent the `status` field for many models. Each status has a human-readable `name` field (e.g. `Active`), and a `slug` field (e.g. `active`).
 
+### Display name
+
+Several models such as device type and VLAN expose a `display_name` property which has been renamed to `display`. In fact, there are several other instances, especially in the REST API, where the `display_name` field is used and as such, all instances have been renamed to `display`.
```
Comment (glennmatthews) [inline nautobot/docs/installation/migrating-from-netbox.md:76]:
```suggestion
Several models such as device type and VLAN exposed a `display_name` property, which has now been renamed to `display`. In fact, there are several other instances, especially in the REST API, where the `display_name` field was used and as such, all instances have been renamed to `display`.
```

**Review by glennmatthews — APPROVED** [review]
Looks terrific!

**Review by jathanism — APPROVED** [review]
YOLO

**Pull request merged**